### PR TITLE
VideoPlayer: make m_pInputStream a smart pointer

### PR DESCRIPTION
**Pull request by lrusak**
as requested by @FernetMenta 

build and run tested on linux x11

### CODE REVIEW

**Comment by lrusak** [comment]
jenkins build this please

**Comment by moralesfelipe** [comment]
This repository appears on Topic mysql

File: xbmc/cores/VideoPlayer/VideoPlayer.cpp [hunk 1]
```diff
@@ -2467,7 +2466,6 @@ void CVideoPlayer::OnExit()
   SAFE_DELETE(m_pDemuxer);
   SAFE_DELETE(m_pSubtitleDemuxer);
   SAFE_DELETE(m_pCCDemuxer);
-  SAFE_DELETE(m_pInputStream);
```
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/VideoPlayer.cpp]: At ecery location vp relases inputsteam I want to make sure that is gets destructed. You could check usecount for being less or equal 1 before reset.

File: xbmc/cores/paplayer/VideoPlayerCodec.cpp [hunk 2]
```diff
@@ -127,8 +124,6 @@ bool VideoPlayerCodec::Init(const CFileItem &file, unsigned int filecache)
     m_pDemuxer = CDVDFactoryDemuxer::CreateDemuxer(m_pInputStream);
     if (!m_pDemuxer)
     {
-      delete m_pInputStream;
-      m_pInputStream = NULL;
```
Comment (FernetMenta) [inline xbmc/cores/paplayer/VideoPlayerCodec.cpp]: Input stream ahould be released here

File: xbmc/cores/paplayer/VideoPlayerCodec.cpp [hunk 3]
```diff
@@ -298,12 +289,6 @@ void VideoPlayerCodec::DeInit()
     m_pDemuxer = NULL;
   }
 
-  if (m_pInputStream != NULL)
```
Comment (FernetMenta) [inline xbmc/cores/paplayer/VideoPlayerCodec.cpp]: Smart pointer are convenient but we should not blindly rely on them. Explicitly release the ref

File: xbmc/cores/VideoPlayer/VideoPlayer.cpp [hunk 4]
```diff
@@ -2466,6 +2466,7 @@ void CVideoPlayer::OnExit()
   SAFE_DELETE(m_pDemuxer);
   SAFE_DELETE(m_pSubtitleDemuxer);
   SAFE_DELETE(m_pCCDemuxer);
+  m_pInputStream.reset();
```
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/VideoPlayer.cpp]: what I mean is:

```
if (m_pInputStream.use_count() > 1)
  throw std::runtime_error("some error");
m_pInputStream.reset;
```

File: xbmc/cores/VideoPlayer/VideoPlayer.cpp [hunk 5]
```diff
@@ -775,8 +775,7 @@ void CVideoPlayer::OnStartup()
 
 bool CVideoPlayer::OpenInputStream()
 {
-  if(m_pInputStream)
-    SAFE_DELETE(m_pInputStream);
+  m_pInputStream.reset();
```
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/VideoPlayer.cpp]: please check use_count here too

**Review by FernetMenta — APPROVED** [review]

**Comment by lrusak** [comment]
jenkins build this please

**Comment by lrusak** [comment]
jenkins build this please

**Comment by Rechi** [comment]
@lrusak fixup commits should have been squashed before merging

**Pull request merged**